ygen/genstate.go

```diff
@@ -373,13 +376,11 @@ func (s *enumGenState) identityrefBaseTypeFromIdentity(i *yang.Identity, noUnder
 	return uniqueName
 }
 
-// resolveEnumName takes a yang.Entry and resolves its name into the type name
-// that will be used in the generated code. Whilst a leaf may only be used
-// in a single context (i.e., at its own path), resolveEnumName may be called
-// multiple times, and hence de-duplication of unique name generation is required.
-// If noUnderscores is set to true, then underscores are omitted from the
-// output name.
-func (s *enumGenState) resolveEnumName(e *yang.Entry, compressPaths, noUnderscores bool) string {
+// enumIdentifier takes in an enum entry and returns a unique identifier for
+// that enum constructed using its path. This identifier would be the same for
+// an enum that's used in two different places in the schema.
+func enumIdentifier(e *yang.Entry, compressPaths bool) string {
+	definingModName := genutil.ParentModulePrettyName(e.Node)
 	// It is possible, given a particular enumerated leaf, for it to appear
 	// multiple times in the schema. For example, through being defined in
 	// a grouping which is instantiated in two places. In these cases, the
```

Comment:
As above, why is it two usages will have the same `yang.Node`? Thanks.

Reply:
When goyang is parsing a YANG schema, then it goes through and finds the nodes that are defined in the schema - which is in relation to what is actually written in the YANG file. When it converts this tree to entries, it then builds the tree such that the schema tree is expanded out -- so if we have:

```yang
grouping foo {
   leaf a { type enumeration { enum A; } }
}

container one { uses foo; }
container two { uses foo; }
```

then there is exactly one `yang.Node` which represents the `a` defined in the `foo` grouping, but there are two entries for `/one/a` and `/two/a`. 

Prior to this change we always de-duplicated these enumerated types, so that there is an enumeration that represents `/foo/a` -- this makes sense in OpenConfig for example, where we might have `/foo/state/an-enum` and `/foo/config/an-enum`, and we don't really want to refer to those with different types. In other schemas, like the example above, de-duplicating these means that we would choose one type for `/one/a` and `/two/a`. Since these types are named after the schema path, then this meant that we had very confusing naming for the enumerations.

Reply:
Thanks! I think I got it now. The most time-consuming parts were
1. Understanding the tests.
2. Finally deciding that some inputs are not worth testing since they're just pass-through parameters to lower-level functions.